### DISCUSSION TITLE
[Flutter 3.10] SearchAnchor.bar() and riverpod

### DISCUSSION THREAD

**Discussion by ips-jm** [rrousselGit/riverpod#2551] [Q&A]
Hey,
has anyone tried the new SearchAnchor.bar() with riverpod yet?
I'm using a AsyncNotifierProvider to call an API with the given search input (.family modifier). The displayed anchor with the suggestions only gets updated when I click out of the text field and get back into it again.

**Comment by cisnux-seed** [comment]
me too, I have implemented Riverpod with a SearchAnchor from Material 3 but don't know why, the list of suggestions wasn't updated correctly.

**Comment by cisnux-seed** [comment]
> Hey, has anyone tried the new SearchAnchor.bar() with riverpod yet? I'm using a AsyncNotifierProvider to call an API with the given search input (.family modifier). The displayed anchor with the suggestions only gets updated when I click out of the text field and get back into it again.

I've fixed it, using Consumer.
```
SearchAnchor(
                      searchController: _searchController,
                      isFullScreen: false,
                      viewBuilder: (suggestions) => Consumer(
                        builder: (_, ref, __) =>
                            ref.watch(suggestionsProvider).whenOrNull(
                                  data: (suggestions) => ListView.builder(
                                    itemCount: suggestions.length,
                                    itemBuilder: (_, index) => ListTile(
                                      title: Text(suggestions[index]),
                                      onTap: () => _searchController.closeView(
                                        suggestions[index],
                                      ),
                                    ),
                                  ),
                                ) ??
                            const SizedBox(),
                      ),
                      suggestionsBuilder: (_, controller) => [],
);
```
